src/full/Agda/TypeChecking/InstanceArguments.hs

```diff
@@ -505,7 +511,6 @@ checkCandidates m t cands =
         where
           runCandidateCheck check =
             flip catchError handle $
-            nowConsideringInstance $
```

Comment:
Moved this to before the call to `checkCandidates` above. I don't think this is necessary, but probably couldn't hurt? It seemed more semantically correct, and the change it makes to the typechecking state could be useful for determining when to hold out constraints.

Reply:
Hmm, just noticed that `stConsideringInstance` is referenced in `reduceTm` in `Typechecking.Reduce.Fast`, with the following comment:

> Are we currently instance searching. In that case we don't fail hard on missing clauses. This is a (very unsatisfactory) work-around for #3870.

Not sure if this change would affect that at all.